### PR TITLE
Syncronize with pggb, more testing

### PR DESCRIPTION
**Pull request by subwaystation**
## PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] `CHANGELOG.md` is updated
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] Documentation in `docs` is updated
- [ ] If necessary, also make a PR on the [nf-core/pangenome branch on the nf-core/test-datasets repo](https://github.com/nf-core/test-datasets/pull/new/nf-core/pangenome)
 
This PR is trying to solve #26.
This PR affects `Dockerfile`, `main.nf`, and `nextflow.config`.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -53,5 +57,91 @@ jobs:
         # TODO nf-core: You can customise CI pipeline run tests as required
         # For example: adding multiple test runs with different parameters
         # Remember that you can parallelise this by using strategy.matrix
+        # We also test basic visualization and reporting options here
         run: |
-          nextflow run ${GITHUB_WORKSPACE} -profile test,docker
+          nextflow run ${GITHUB_WORKSPACE} -profile test,docker ${{ matrix.parameters }}
+  edyeet:
+      name: Test edyeet with workflow parameters
+      if: ${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/rnaseq') }}
```
Comment (Zethson) [inline .github/workflows/ci.yml:65]: cough rnaseq cough

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -53,5 +57,91 @@ jobs:
...
+        - name: Install Nextflow
+          run: |
+            wget -qO- get.nextflow.io | bash
+            sudo mv nextflow /usr/local/bin/
+        - name: Run pipeline with edyeet and various parameters
+          run: |
+            nextflow run ${GITHUB_WORKSPACE} -profile test,docker ${{ matrix.parameters }}
+  seqwish:
+      name: Test seqwish with workflow parameters
+      if: ${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/rnaseq') }}
```
Comment (Zethson) [inline .github/workflows/ci.yml:94]: same

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -53,5 +57,91 @@ jobs:
...
+        - name: Install Nextflow
+          run: |
+            wget -qO- get.nextflow.io | bash
+            sudo mv nextflow /usr/local/bin/
+        - name: Run pipeline with seqwish and various parameters
+          run: |
+            nextflow run ${GITHUB_WORKSPACE} -profile test,docker ${{ matrix.parameters }}
+  smoothxg:
+      name: Test smoothxg with workflow parameters
+      if: ${{ github.event_name != 'push' || (github.event_name == 'push' && github.repository == 'nf-core/rnaseq') }}
```
Comment (Zethson) [inline .github/workflows/ci.yml:119]: please

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -53,5 +57,91 @@ jobs:
...
+          uses: actions/checkout@v2
+
+        - name: Install Nextflow
+          run: |
+            wget -qO- get.nextflow.io | bash
+            sudo mv nextflow /usr/local/bin/
+        - name: Run pipeline with smoothxg and various parameters
+          run: |
+            nextflow run ${GITHUB_WORKSPACE} -profile test,docker ${{ matrix.parameters }}                        
+
```
Comment (Zethson) [inline .github/workflows/ci.yml:147]: EOL sir

**Comment by subwaystation** [comment]
Hmm, if you take a look at https://github.com/nf-core/pangenome/pull/29/checks?check_run_id=1487357914, you can see that it does both `--do_layout` and `--do_viz`, although only one of them is specified. Any ideas why?!

File: main.nf [hunk 5]
```diff
@@ -11,18 +11,39 @@
 
 nextflow.enable.dsl = 2
 
+// We can't change global parameters inside this scope, so we build the ones we need locally
+def edyeet_merge_cmd=params.edyeet_merge_cmd
```
Comment (Zethson) [inline main.nf:15]: Maybe what works is the following:

Rewrite the parameter definitions below into:
`edyeet_merge_cmd = "-M" ? params.edyeet_merge_cmd : ''`

You can apply such a pattern to all of them.

Comment (Zethson) [reply main.nf:15]: Replace the stuff behind the colon with your default of choice.

**Review by Zethson — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 4 above)
Comment (subwaystation) [reply .github/workflows/ci.yml:147]: I have an empty line here. What is it you want to tell me?

Comment (Zethson) [reply .github/workflows/ci.yml:147]: The Github diff told me that there's an EOL missing.
If there is one let's ignore it :)

File: main.nf (hunk 5 above)
Comment (subwaystation) [reply main.nf]: Thanks for the ternary operator tip. The actual code looks a little bit different, though. Please see for yourself.

Comment (Zethson) [reply main.nf]: image

File: .github/workflows/ci.yml [hunk 6]
```diff
@@ -144,4 +144,3 @@ jobs:
         - name: Run pipeline with smoothxg and various parameters
           run: |
             nextflow run ${GITHUB_WORKSPACE} -profile test,docker ${{ matrix.parameters }}                        
-
```
Comment (Zethson) [inline .github/workflows/ci.yml]: Now there really is no EOL :)
I verified it.

**Review by Zethson — APPROVED** [review]
LGTM

File: .github/workflows/ci.yml (hunk 6 above)
Comment (subwaystation) [reply .github/workflows/ci.yml]: image
My editor says otherwise, hmmmmmm

Comment (subwaystation) [reply .github/workflows/ci.yml]: This is resolved now. I don't know why Visual Studio is lying to me.

**Pull request merged**